src/sinol_make/helpers/package_util.py

```diff
@@ -143,3 +152,30 @@ def get_memory_limit(test_path, config, lang, args=None):
 
     str_config = util.stringify_keys(config)
     return _get_limit(LimitTypes.MEMORY_LIMIT, test_path, str_config, lang)
+
+
+def validate_files(task_id):
+    """
+    Checks if all files in the package have valid names.
+    """
+    def get_invalid_files(path, *patterns):
+        invalid_files = []
+        for file in glob.glob(os.path.join(os.getcwd(), path)):
+            invalid = True
+            for pattern in patterns:
+                if pattern.match(os.path.basename(file)):
+                    invalid = False
+                    break
+            if invalid:
+                invalid_files.append(os.path.basename(file))
+        return invalid_files
+
+    in_test_re = re.compile(r'^(%s(([0-9]+)([a-z]?[a-z0-9]*))).in$' % (re.escape(task_id)))
```

Comment:
In general, I find regex quite unreadable. It takes a lot of time to understand this line of code and even after some time I'm not sure whether it is correct. Can we make some simple code that does what this regex tries to do, but is readable and hard to make a bug?

Reply:
I copied this regex from sio's code and I don't think that changing this regex to a function will be more readable

Reply:
So what does this regex do, in detail?

Reply:
Test name has to begin with task id, then at least one digit, then an optional letter, then any string consisting of letters and digits